models/articles.model.js

```diff
@@ -21,21 +21,41 @@ const sortByOptions = [
       if (!sortByOptions.includes(sort_by) ||!orderOptions.includes(order) ) {
         return Promise.reject({ status: 400, msg: 'Not Valid' })   
       }
-    
-      
-    return db.query(`
-        SELECT articles.article_id, articles.author, articles.title, articles.topic, 
-               articles.created_at, articles.votes, articles.article_img_url, 
-               COUNT(comments.article_id)::INT AS comment_count
-        FROM articles
-        LEFT JOIN comments ON comments.article_id = articles.article_id
-        GROUP BY articles.article_id
-        ORDER BY ${sort_by} ${order.toUpperCase()};
-      `)
+    const queryParams = [];
+  
+    if (topic) {
+      let queryForArticle = `
+      SELECT 
+        articles.article_id, articles.author, articles.title, articles.topic, 
+        articles.created_at, articles.votes, articles.article_img_url, 
+        COUNT(comments.article_id)::INT AS comment_count
+      FROM articles
+      LEFT JOIN comments ON comments.article_id = articles.article_id
+      WHERE articles.topic = $1
+    GROUP BY articles.article_id
+    ORDER BY ${sort_by} ${order.toUpperCase()}
+    `;
+      queryParams.push(topic)
+    }
+    else{queryForArticle=`
+      SELECT 
+        articles.article_id, articles.author, articles.title, articles.topic, 
+        articles.created_at, articles.votes, articles.article_img_url, 
+        COUNT(comments.article_id)::INT AS comment_count
+      FROM articles
+      LEFT JOIN comments ON comments.article_id = articles.article_id
+    GROUP BY articles.article_id
+    ORDER BY ${sort_by} ${order.toUpperCase()}
```

Comment:
This is not very DRY, see if you can find a way to build up one query string conditionally.